test/net/sourceforge/kolmafia/request/RelayRequestTest.java

```diff
@@ -80,6 +85,57 @@ public void exerciseSomeObjectMethodsForCoverage() {
     rr.constructURLString("diary.php?textversion=1");
   }
 
+  @Nested
+  class LocalFiles {
+    private RelayRequest makeFileRequest(String path) throws IOException {
+      Files.copy(Paths.get("request", path), Paths.get("relay", path));
+      var rr = new RelayRequest(true);
+      rr.constructURLString(path, true);
+      rr.run();
+      Files.deleteIfExists(Paths.get("relay", path));
+      return rr;
+    }
+
+    @BeforeAll
+    public static void beforeAll() throws IOException {
+      try {
+        Files.createDirectory(Paths.get("relay"));
+      } catch (FileAlreadyExistsException e) {
+      }
+    }
+
+    @Test
+    public void returnsNotFound() {
+      var rr = new RelayRequest(true);
+      rr.constructURLString("nonexistent.png", true);
+      rr.run();
+
+      assertThat(rr.statusLine, is("HTTP/1.1 404 Not Found"));
+      assertThat(rr.responseCode, is(404));
+    }
+
+    @ParameterizedTest
+    @ValueSource(
+        strings = {
+          "test_relay_request_text.txt",
+          "test_relay_request_html.html",
+        })
+    public void returnsTextFile(String filename) throws IOException {
+      var rr = makeFileRequest(filename);
+      assertThat(rr.statusLine, is("HTTP/1.1 200 OK"));
+      assertThat(rr.responseCode, is(200));
+      assertThat(rr.responseText.trim(), is(html("request/" + filename).trim()));
+    }
+
+    @Test
+    public void returnsPng() throws IOException {
+      var rr = makeFileRequest("test_relay_request_sample.png");
```

Comment:
this png is apparently not a valid png as far as GitHub is concerned... could you make it valid?